lib/extensions/ar_virtual.rb

```diff
@@ -111,6 +111,9 @@ def virtual_delegate(*methods)
     #
     # this is called at schema load time (and not at class definition time)
     #
+    # Of note, klass.type_for_attribute() loads the schema of the target type.
+    # In rare cases, this can cause race conditions
```

Comment:
is this due to circular dependencies?

Reply:
probably not. the bug says there are circular dependencies